### ISSUE TITLE
Unable to Install on PopOS 20.04

### ISSUE THREAD

**Issue by jeremyakers** [paradoxxxzero/gnome-shell-system-monitor-applet#774]
I'm trying to install the system monitor applet. I've tried installing via both Chrome (109.0.5414.74) and Firefox (108.0.2)
First thing I notice both browsers show this warning even before I attempted to install:
![Screenshot 2023-01-20 at 14-22-06 system-monitor - GNOME Shell Extensions](https://user-images.githubusercontent.com/13689422/213798728-d8bae0c8-deb1-4408-8009-5c86e52ca712.png)

In an attempt to resolve this, before even installing: I check the docs at the specified link which say to install `gnome-chrome-shell`:
https://wiki.gnome.org/action/show/Projects/GnomeShellIntegration/Installation?action=show&redirect=Projects%2FGnomeShellIntegrationForChrome%2FInstallation#Ubuntu_Linux
Like so:
![Screenshot 2023-01-20 at 14-23-56 Projects_GnomeShellIntegration_Installation - GNOME Wiki!](https://user-images.githubusercontent.com/13689422/213798878-565c1ea9-215d-4a1f-8f67-b8f32b920f6a.png)

Which I already had installed:
```
$ dpkg -l chrome-gnome-shell
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name               Version      Architecture Description
+++-==================-============-============-===================================================
ii  chrome-gnome-shell 10.1-5       all          GNOME Shell extensions integration for web browsers
```

I have the browser plugin installed for both Chrome and Firefox:
![Screenshot from 2023-01-20 14-27-36](https://user-images.githubusercontent.com/13689422/213799476-9cdc0862-d169-477e-bac8-32a5fd44a296.png)

So since I seem to have all the required components installed: I went ahead and tried to install the system-monitor extension. Firefox prompts me for permission to install, which I grant, and it just says "ERROR":
![Screenshot 2023-01-20 at 14-11-48 system-monitor - GNOME Shell Extensions](https://user-images.githubusercontent.com/13689422/213799876-1aa63213-b18b-4023-9321-b77103d80d85.png)

If I refresh the page it adds the button to configure the extension:
![Screenshot 2023-01-20 at 14-30-57 system-monitor - GNOME Shell Extensions](https://user-images.githubusercontent.com/13689422/213800008-2daff8ba-a753-43ec-88af-d9d1d582813a.png)

I'm able to open the preferences / configuration GUI:
![Screenshot from 2023-01-20 14-12-15](https://user-images.githubusercontent.com/13689422/213800472-15e1c074-21cd-480d-bdd3-9e9090d7626f.png)

Running `gnome-extensions` info/show just shows:
```
$ gnome-extensions show system-monitor@paradoxxx.zero.gmail.com
system-monitor@paradoxxx.zero.gmail.com
  Name: system-monitor
  Description: Display system information in GNOME Shell status bar, such as memory, CPU, disk and battery usages, network rates…
  Path: /home/jeremy/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com
  URL: https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet
  Version: 40
  State: ERROR
$ gnome-extensions info system-monitor@paradoxxx.zero.gmail.com
system-monitor@paradoxxx.zero.gmail.com
  Name: system-monitor
  Description: Display system information in GNOME Shell status bar, such as memory, CPU, disk and battery usages, network rates…
  Path: /home/jeremy/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com
  URL: https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet
  Version: 40
  State: ERROR
```

I have the pre-reqs installed:
```
$ sudo apt install gir1.2-gtop-2.0 gir1.2-nm-1.0 gir1.2-clutter-1.0 gnome-system-monitor
Reading package lists... Done
Building dependency tree       
Reading state information... Done
gir1.2-clutter-1.0 is already the newest version (1.26.4+dfsg-1).
gir1.2-gtop-2.0 is already the newest version (2.40.0-2).
gir1.2-nm-1.0 is already the newest version (1.22.10-1ubuntu2.3).
gnome-system-monitor is already the newest version (3.36.1-0ubuntu0.20.04.1).

```

I also tried downloading the ZIP and installing/enabling via `gnome-extensions` CLI.

I'm lost as to how to troubleshoot this any further. How do I access an actual log/stacktrace/etc for why the extension is failing to load?

**Comment by jeremyakers** [comment]
Updated to add: 
I installed gnome-shell-extensions-manager and that gave me access to the underlying error message:
`Error: Tried to construct an object without a GType; are you using GObject.registerClass() when inheriting from a GObject type?`

**Comment by jeremyakers** [comment]
After installing gnome-shell-extensions-manager, and using that to finally get at the true underlying error message: I searched the issues here on github and found the solution:

`Alt-F2` then `r`

This reloaded gnome-shell and fixed it for me.

**Comment by ZimbiX** [comment]
I'm glad you figured that out. To see error logging:

```bash
journalctl -f /usr/bin/gnome-shell /usr/bin/gjs
```

For more up-to-date Gnome Shell versions, you'd need to instead install [system-monitor-next](https://extensions.gnome.org/extension/3010/system-monitor-next/) due to delays from the current maintainers of this repo

**Comment by ZimbiX** [comment]
I think this issue can be closed now. Needing to reload after installing is common to many Gnome extensions. I don't yet know why it's the case here. If it's going to be addressed, there should be an issue dedicated to that, if it's still a problem with system-monitor-next.

**Comment by ZimbiX** [comment]
Oh, and I haven't seen that warning before. Maybe it's because your distro's Gnome packages are older than what the website / browser extension is expecting...

**Comment by jeremyakers** [comment]
Yeah I guess my only feedback would be to add a step in the documentation that says reloading may be necessary (And explain how). I've personally never had to reload for any other extensions I've installed, and even though I've been a GNOME user for many, many years I did not even know about that reload command.

I think it's fine that reload may be necessary, but a user may not know that it's necessary nor how to do it.
